### PR TITLE
Convert MIDI labels to STL map

### PR DESCRIPTION
**Pull request by stellar-aria**
Also converts the backing label storage of a String to a std::string_view, since only the raw C-string was ever stored/retrieved

### CODE REVIEW

File: src/deluge/model/instrument/midi_instrument.cpp [hunk 1]
```diff
@@ -1237,19 +1237,25 @@ void MIDIInstrument::sendNoteToInternal(bool on, int32_t note, uint8_t velocity,
 	}
 }
 
-String* MIDIInstrument::getNameFromCC(int32_t cc) {
+std::string_view MIDIInstrument::getNameFromCC(int32_t cc) {
 	if (cc >= 0 && cc < kNumRealCCNumbers) {
-		MIDILabel* midiLabel = midiLabelCollection.labels.getLabelFromCC(cc);
```
Comment (seangoodvibes) [inline src/deluge/model/instrument/midi_instrument.cpp:1242]: Looking at this on my phone and this change seems wrong here as if it's >=0 and < num real cc's then it's in range, not out of range

Will double check when I get in front of computer

File: src/deluge/model/instrument/midi_instrument.h [hunk 2]
```diff
@@ -137,5 +137,6 @@ class MIDIInstrument final : public NonAudioInstrument {
 	void outputAllMPEValuesOnMemberChannel(int16_t const* mpeValuesToUse, int32_t outputMemberChannel);
 	Error readMIDIParamFromFile(Deserializer& reader, int32_t readAutomationUpToPos,
 	                            MIDIParamCollection* midiParamCollection, int8_t* getCC = NULL);
-	MIDILabelCollection midiLabelCollection;
+
+	deluge::fast_map<uint8_t, std::string_view> labels;
```
Comment (seangoodvibes) [inline src/deluge/model/instrument/midi_instrument.h:141]: Just want to confirm, does this change have any impact on the weight of the label strings in memory?

I know it was one of the reasons Mark pushed me to use a resizable array.

Comment (stellar-aria) [reply src/deluge/model/instrument/midi_instrument.h:141]: No if anything, it's lighter.

File: src/deluge/model/instrument/midi_instrument.cpp (hunk 1 above)
Comment (stellar-aria) [reply src/deluge/model/instrument/midi_instrument.cpp:1242]: Fixed

**Review by seangoodvibes — APPROVED** [review]

**Pull request merged**